src/astro/files/__init__.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pathlib
+from typing import List, Optional
+
+import pandas as pd
+import smart_open
+
+from astro.constants import LOAD_DATAFRAME_BYTES_LIMIT, FileType
+from astro.files.locations import location_factory
+from astro.files.type import type_factory
+from astro.utils.file import get_size
+
+
+class File:
+    def __init__(self, path: str, conn_id: Optional[str] = None):
+        """Init file object which represent a single file in local/object stores
+
+        :param path: Path to a file in the filesystem/Object stores
+        :param conn_id: Airflow connection ID
+        """
+        self.location = location_factory(path, conn_id)
+        self.type = type_factory(path)
+
+    @property
+    def path(self):
+        return self.location.path
+
+    def get_size(self) -> int:
+        """
+        Return the size (bytes) of the given file.
+
+        :return: File size in bytes
+        :rtype: int
+        """
+        path = pathlib.Path(self.path)
+        return os.path.getsize(path)
```

Comment:
Wouldn't this only work for local files?

Reply:
Or is the idea that this is how we handle the file once it has been loaded?

Reply:
We need an implementation for all file locations, for now, I'll just move this to Local.py with the base class having the size method. I'll create a separate PR for this.